### PR TITLE
Add metrics for DaemonSet rolling updates

### PR DESCRIPTION
**Pull request by janetkuo**
@erictune @kargakis @lukaszo @kubernetes/sig-apps-pr-reviews 

```release-note
```

### CODE REVIEW

File: pkg/controller/daemon/metrics_cache.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
+package daemon
+
+import (
+	"hash/adler32"
```
Comment (kargakis) [inline pkg/controller/daemon/metrics_cache.go]: use fnv

File: pkg/controller/daemon/metrics.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+import (
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	DaemonSetUpdateSubsystem = "daemonset_rollingupdate"
```
Comment (kargakis) [inline pkg/controller/daemon/metrics.go]: unexport these

File: pkg/controller/daemon/metrics_cache.go (hunk 1 above)
Comment (janetkuo) [inline pkg/controller/daemon/metrics_cache.go]: Updated

File: pkg/controller/daemon/metrics.go (hunk 2 above)
Comment (janetkuo) [inline pkg/controller/daemon/metrics.go]: done

**Comment by lukaszo** [comment]
How Can I test it?

**Comment by janetkuo** [comment]
@lukaszo this is still WIP since it depends on the DS rolling update PR

**Comment by janetkuo** [comment]
@kargakis PTAL

**Comment by lukaszo** [comment]
@janetkuo Is there somewhere a description how to configure the metrics?

**Comment by janetkuo** [comment]
Added a unit test. 

@lukaszo The metrics is configured by the DaemonSet controller. The idea is that it used `podMetrics` instances to keep track of pods deletion/creation on a node of each DaemonSet, and use that to calculate the time between the pod being deleted, recreated, and become ready. These metrics are registered with the Prometheus registry and exposed in endpoint "/metrics".

**Comment by ethernetdan** [comment]
Removing issue from 1.6 milestone since code freeze has begun:
* If this is a bug fix, please label as `kind/bug` and reapply the milestone.
* If this is a feature for 1.6, an [exception](https://github.com/kubernetes/features/blob/master/EXCEPTIONS.md) should be filed for it before 11:00PM PST on 3/2. Note that most exceptions to code freeze are not accepted.
* If this is a feature for 1.7, feel free to add the 1.7 milestone to the PR.

/cc @kubernetes/release-team @kubernetes/kubernetes-release-managers

**Comment by calebamiles** [comment]
Moving this issue to the 1.7 milestone @janetkuo.

**cc: @ethernetdan, @kubernetes/kubernetes-release-managers**

**Pull request closed without merging**